pkg/deploy/api/v1/defaults.go

```diff
@@ -100,8 +100,10 @@ func SetDefaults_RollingDeploymentStrategyParams(obj *RollingDeploymentStrategyP
 func SetDefaults_DeploymentConfig(obj *DeploymentConfig) {
 	for _, t := range obj.Spec.Triggers {
 		if t.ImageChangeParams != nil {
-			t.ImageChangeParams.From.Kind = "ImageStreamTag"
```

Comment:
So don't always default, but validation will fail if its not that value?  I can live with that and validation was already that tight as I recall.

However, didn't we have this running unconditionally in order to force the transformation of old data?